test/services/delete_bill_run.service.test.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+const Sinon = require('sinon')
+
+const { describe, it, beforeEach, afterEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const {
+  AuthorisedSystemHelper,
+  BillRunHelper,
+  DatabaseHelper,
+  GeneralHelper,
+  RegimeHelper
+} = require('../support/helpers')
+
+const {
+  BillRunModel,
+  InvoiceModel,
+  LicenceModel,
+  TransactionModel
+} = require('../../app/models')
+
+const { presroc: requestFixtures } = require('../support/fixtures/create_transaction')
+const { presroc: chargeFixtures } = require('../support/fixtures/calculate_charge')
+
+const { rulesService: rulesServiceResponse } = chargeFixtures.simple
+
+const { CreateTransactionService } = require('../../app/services')
+
+// Things we need to stub
+const { RulesService } = require('../../app/services')
+
+// Thing under test
+const { DeleteBillRunService } = require('../../app/services')
+
+describe('Delete Bill Run service', () => {
+  let billRun
+  let authorisedSystem
+  let regime
+  let payload
+
+  beforeEach(async () => {
+    await DatabaseHelper.clean()
+    regime = await RegimeHelper.addRegime('wrls', 'WRLS')
+    authorisedSystem = await AuthorisedSystemHelper.addSystem('1234546789', 'system1', [regime])
+
+    // We clone the request fixture as our payload so we have it available for modification in the invalid tests. For
+    // the valid tests we can use it straight as
+    payload = GeneralHelper.cloneObject(requestFixtures.simple)
+
+    Sinon.stub(RulesService, 'go').returns(rulesServiceResponse)
+    billRun = await BillRunHelper.addBillRun(authorisedSystem.id, regime.id)
+
+    await CreateTransactionService.go(payload, billRun, authorisedSystem, regime)
+  })
+
+  afterEach(async () => {
+    Sinon.restore()
+  })
+
+  describe('When a valid bill run is supplied', () => {
+    it('deletes the bill run', async () => {
+      await DeleteBillRunService.go(billRun)
+
+      const result = await BillRunModel.query().findById(billRun.id)
+      expect(result).to.not.exist()
+    })
+
+    it('deletes the bill run invoices', async () => {
+      await DeleteBillRunService.go(billRun)
+
+      const invoices = await InvoiceModel.query().select().where({ billRunId: billRun.id })
+      expect(invoices).to.be.empty()
+    })
+
+    it('deletes the bill run licences', async () => {
+      await DeleteBillRunService.go(billRun)
+
+      const licences = await LicenceModel.query().select().where({ billRunId: billRun.id })
+      expect(licences).to.be.empty()
+    })
+
+    it('deletes the bill run transactions', async () => {
+      await DeleteBillRunService.go(billRun)
+
+      const transactions = await TransactionModel.query().select().where({ billRunId: billRun.id })
+      expect(transactions).to.be.empty()
+    })
+  })
```

Comment:
Thanks to the fantastic `TransactionHelper` you added awhile back I think you can do a bunch to simplify the tests. There are a few things to cover, so I figured it'd just be easier to show rather than tell. With this we don't need to reference any models or services, nor do we need to mock anything or use the fixtures directly.

Hope it makes sense. I've also tried it locally and the tests all still pass 😁 

```suggestion

const { describe, it, beforeEach } = exports.lab = Lab.script()
const { expect } = Code

// Test helpers
const {
  BillRunHelper,
  DatabaseHelper,
  GeneralHelper,
  TransactionHelper
} = require('../support/helpers')

// Thing under test
const { DeleteBillRunService } = require('../../app/services')

describe('Delete Bill Run service', () => {
  let billRun

  beforeEach(async () => {
    await DatabaseHelper.clean()

    billRun = await BillRunHelper.addBillRun(GeneralHelper.uuid4(), GeneralHelper.uuid4())

    await TransactionHelper.addTransaction(billRun.id)
  })

  describe('When a valid bill run is supplied', () => {
    it('deletes the bill run', async () => {
      await DeleteBillRunService.go(billRun)

      const refreshedBillRun = await billRun.$query()

      expect(refreshedBillRun).to.not.exist()
    })

    it('deletes the bill run invoices', async () => {
      await DeleteBillRunService.go(billRun)

      const invoices = await billRun.$relatedQuery('invoices')

      expect(invoices).to.be.empty()
    })

    it('deletes the bill run licences', async () => {
      await DeleteBillRunService.go(billRun)

      const licences = await billRun.$relatedQuery('licences')

      expect(licences).to.be.empty()
    })

    it('deletes the bill run transactions', async () => {
      await DeleteBillRunService.go(billRun)

      const transactions = await billRun.$relatedQuery('transactions')

      expect(transactions).to.be.empty()
    })
  })
```